src/core/lib/support/arena.cc

```diff
@@ -22,6 +22,45 @@
 #include <grpc/support/log.h>
 #include <grpc/support/useful.h>
 
+//#define SIMPLE_ARENA_FOR_DEBUGGING
+
+#ifdef SIMPLE_ARENA_FOR_DEBUGGING
```

Comment:
I don't like custom defines... They get stale since they are never tested until needed and they clutter the code. Does this simple arena give significantly better debug-ability? Could we have this simply be the default for ifndef NDEBUG?

Reply:
This was something suggested by @ctiller at one point while I was trying to debug something.  It's been long enough that I don't actually remember why I needed it at this point, but I remember Craig saying that it might be useful for other things in the future, so I left it in.  If we don't think it's going to be useful, I am happy to remove it, or even move it to a separate PR if you want.